scripts/serverless.js

```diff
@@ -495,24 +497,38 @@ const processSpanPromise = (async () => {
 
     const configurationFilename = configuration && configurationPath.slice(serviceDir.length + 1);
 
-    if (isInteractiveSetup) {
-      require('../lib/cli/ensure-supported-command')(configuration);
+    const isStandaloneCommand = standaloneCommands.includes(command);
 
-      if (!process.stdin.isTTY && !process.env.SLS_INTERACTIVE_SETUP_ENABLE) {
-        throw new ServerlessError(
-          'Attempted to run an interactive setup in non TTY environment.\n' +
-            "If that's intentended enforce with SLS_INTERACTIVE_SETUP_ENABLE=1 environment variable",
-          'INTERACTIVE_SETUP_IN_NON_TTY'
-        );
-      }
-      const { configuration: configurationFromInteractive } =
-        await require('../lib/cli/interactive-setup')({
+    if (isInteractiveSetup || isStandaloneCommand) {
+      let configurationFromInteractive;
+
+      if (isInteractiveSetup) {
+        require('../lib/cli/ensure-supported-command')(configuration);
+
+        if (!process.stdin.isTTY && !process.env.SLS_INTERACTIVE_SETUP_ENABLE) {
+          throw new ServerlessError(
+            'Attempted to run an interactive setup in non TTY environment.\n' +
+              "If that's intentended enforce with SLS_INTERACTIVE_SETUP_ENABLE=1 environment variable",
+            'INTERACTIVE_SETUP_IN_NON_TTY'
+          );
+        }
+        const result = await require('../lib/cli/interactive-setup')({
           configuration,
           serviceDir,
           configurationFilename,
           options,
           commandUsage,
         });
+        configurationFromInteractive = result.configuration;
+      } else if (isStandaloneCommand) {
```

Comment:
`isStandaloneCommand` is guaranteed to be truthy here, so I'd leave it as `} else {`

Reply:
Idea was that, in the case we adding more cases into the parent condition block
https://github.com/serverless/serverless/blob/f32b6c346ef34bff7f64f7154dcd3c5ca8ccf729/scripts/serverless.js#L502
This explicit `if (isStandaloneCommand)` would ensure that the other cases don't go into the procedure dedicated to standalone command

On the other hand, it could be premature optimization. Let me leave it as `} else {`

Reply:
> Idea was that, in the case we adding more cases into the parent condition block

I think we'd rather refactor interactive setup, so it can be handled generically instead of introducing other special cases